src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -200,20 +335,27 @@ private static WellKnownMember GetTupleTypeMember(int arity, int position)
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item4,
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item5,
                                                             WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item6,
-                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 }
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 },
+
+                                                        new[]{
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item1,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item2,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item3,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item4,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item5,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item6,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item7,
+                                                            WellKnownMember.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Rest }
         };
 
         /// <summary>
         /// Returns "Item1" for position=0
+        /// Returns "Item12" for position=11
         /// </summary>
-        internal static string UnderlyingMemberName(int position)
+        internal static string TupleMemberName(int position)
         {
-            if (position > 6)
-            {
-                // PROTOTYPE
-                position = 0;
-            }
-            return memberNames[position];
+            var number = position + 1;
+            return "Item" + number;
```

Comment:
It seems Item9, Item10 and so on should not be reserved. They do not clash with anything. 
We should consider allowing Item9, just like we allow ITEM9, item9, Item_9, Item09  etc.. anyways.